File: v1/src/main/java/com/api/v1/domain/Payment.java
package com.api.v1.domain;

import java.math.BigInteger;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.Getter;

@Getter
@Document(collection = "v1_payments")
public class Payment {

    @Id
    private UUID id;

    @Field
    private BigInteger orderNumber;

    @Field
    private Customer customer;

    @Field
    private Car car;

    @Field
    private CreditCard creditCard;

    @Field
    private String paidAt;

    public Payment(
            UUID id,
            BigInteger orderNumber,
            Customer customer,
            Car car,
            CreditCard creditCard,
            String paidAt
    ) {
        this.id = id;
        this.customer = customer;
        this.orderNumber = orderNumber;
        this.car = car;
        this.creditCard = creditCard;
        this.paidAt = paidAt;
    }

}

File: v2/src/main/kotlin/com/api/v2/payment/domain/Payment.kt
package com.api.v2.payment.domain

import com.api.v2.car.domain.Car
import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.customer.domain.Customer
import com.api.v2.payment.utils.PaymentOrderGeneratorUtil
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigInteger
import java.time.ZonedDateTime
import java.util.UUID

@Document(collection = "v2_payments")
data class Payment(
    @Id val id: UUID,
    val paymentOrder: BigInteger,
    val customer: Customer,
    val car: Car,
    val creditCard: CreditCard,
    val paidAt: String
) {

    constructor(
        customer: Customer,
        car: Car,
        creditCard: CreditCard
    ): this (
        UUID.randomUUID(),
        PaymentOrderGeneratorUtil.generate(),
        customer,
        car,
        creditCard,
        ZonedDateTime.now().toString()
    )

}
